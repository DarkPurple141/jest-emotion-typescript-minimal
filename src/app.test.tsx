import React from 'react';
import Component from './app';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';

// Ideal API (doesn't serialize)
test('Ideal', () => {
  expect(<Component />).toMatchDiffSnapshot(<Component isError />);
})

// works except for [object Object] for component
test('Using react-rest-render', () => {
  expect(
    renderer
    .create(<Component />)
    .toJSON()
  ).toMatchDiffSnapshot(renderer
    .create(<Component isError />)
    .toJSON());
})

// works except for [object Object] for component
test('using snapshotDiff() func', () => {
  const treeA =  renderer
    .create(<Component />)
    .toJSON()

  const treeB = renderer
    .create(<Component isError />)
    .toJSON()
  expect(snapshotDiff(treeA, treeB)).toMatchSnapshot()
})
