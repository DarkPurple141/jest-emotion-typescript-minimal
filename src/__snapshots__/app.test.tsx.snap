// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ideal 1`] = `
"Snapshot Diff:
- <App />
+ <App isError={true} />

@@ -3,10 +3,10 @@
      className=\\"css-gj8yp6-paragraphStyles\\"
    >
      Hello
    </p>
    <button
-     className=\\"css-qrvbnj-buttonStyles\\"
+     className=\\"css-1d3nsau-errorStyles\\"
    >
      World
    </button>
  </div>"
`;

exports[`Using react-rest-render 1`] = `
"Snapshot Diff:
- First value
+ Second value

  .emotion-0 {
    color: red;
  }

  .emotion-1 {
-   background-color: blue;
+   background-color: red;
  }

  [object Object]"
`;

exports[`using snapshotDiff() func 1`] = `
"Snapshot Diff:
- First value
+ Second value

  .emotion-0 {
    color: red;
  }

  .emotion-1 {
-   background-color: blue;
+   background-color: red;
  }

  [object Object]"
`;

exports[`vanilla toMatchSnapshot() 1`] = `
.emotion-0 {
  color: red;
}

.emotion-1 {
  background-color: blue;
}

<div>
  <p
    className="emotion-0"
  >
    Hello
  </p>
  <button
    className="emotion-1"
  >
    World
  </button>
</div>
`;
